source.txt

```diff
@@ -166,8 +166,10 @@ Table of Contents
 
         URI_ENCODE( <parent_folder> <folder_name> '/' )
 
-    for a folder. Item names MAY contain all characters except '/' and
-    the null character, and MUST NOT have zero length.
+    for a folder. Item names MUST only consist of unreserved characters,
+    as defined in Section 2.3 of [URI]. Item names MUST NOT be equal to
```

Comment:
I was talking about the reference to the URI RFC:

We have been over this. Reserved characters are NOT okay to use in practice, unencoded or not. I've repeatedly stated that server and proxying software will make the same mess out of both encoded and unencoded reserved characters, and have given nginx as a particular example. This is the whole point of this pull request, to avoid this kind of issue.

I agree that we also need to clarify how item names should be shown in folder listings, but this PR almost solves that issue as a side-effect.